src/transformState.js

```diff
@@ -4,8 +4,34 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
 function transformState(state, actions) {
-  // write code here
+  for (const properties of actions) {
+    const { type, extraData, keysToRemove } = properties;
+
+    switch (type) {
+      case 'addProperties':
+        for (const value in extraData) {
+          state[value] = extraData[value];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const value of keysToRemove) {
+          if (value in state) {
+            delete state[value];
+          }
+        }
+        break;
+
+      default:
+        for (const key in state) {
+          delete state[key];
+        }
```

Comment:
```suggestion
      case 'clear':
        for (const key in state) {
          delete state[key];
        }
        break;
        
      default:
      break;
```

Краще явно прописувати кожен кейс